lib/ansible/modules/monitoring/zabbix/zabbix_host_facts.py

```diff
@@ -0,0 +1,245 @@
+
+    HAS_ZABBIX_API = True
+except ImportError:
+    HAS_ZABBIX_API = False
+
+
+class Host(object):
+    def __init__(self, module, zbx):
+        self._module = module
+        self._zapi = zbx
+
+    def is_host_exist(self, host_name, exact_match):
+        """ Check host exists """
+        search_key = 'search'
+        if exact_match:
+            search_key = 'filter'
+        result = self._zapi.host.get({search_key: {'host': host_name}})
+        return result
+
+    def get_hosts_by_host_name(self, host_name, exact_match):
+        """ Get host by host name """
+        search_key = 'search'
+        if exact_match:
+            search_key = 'filter'
+        host_list = self._zapi.host.get({'output': 'extend', search_key: {'host': [host_name]}})
+        if len(host_list) < 1:
+            self._module.fail_json(msg="Host not found: %s" % host_name)
+        else:
+            return host_list
+
+    def get_hosts_by_ip(self, host_ips):
+        """ Get host by host ip(s) """
+        hostinterfaces = self._zapi.hostinterface.get({
+            'output': 'extend',
+            'filter': {
+                'ip': host_ips
+            }
+        })
+        if len(hostinterfaces) < 1:
+            self._module.fail_json(msg="Host not found: %s" % host_ips)
+        host_list = []
+        for hostinterface in hostinterfaces:
+            host = self._zapi.host.get({
+                'output': 'extend',
+                'selectGroups': 'extend',
+                'hostids': hostinterface['hostid']
+            })
+            host[0]['hostinterfaces'] = hostinterface
+            host_list.append(host[0])
+        return host_list
+
+    def delete_duplicate_hosts(self, hosts):
+        """ Delete duplicated hosts """
+        unique_hosts = []
+        listed_hostnames = []
+        for zabbix_host in hosts:
+            if zabbix_host['name'] in listed_hostnames:
+                self._zapi.host.delete([zabbix_host['hostid']])
+                continue
+            unique_hosts.append(zabbix_host)
+            listed_hostnames.append(zabbix_host['name'])
+        return unique_hosts
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            host_name=dict(type='str', default='', required=False),
+            host_ip=dict(type='list', default=[], required=False),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            timeout=dict(type='int', default=10),
+            exact_match=dict(type='bool', required=False, default=False),
+            remove_duplicate=dict(type='bool', required=False, default=False)
```

Comment:
Maybe have it as `default=True`? Imo it won't have any impact on searches using `host_name` and will prevent duplicate responses when using `host_ip` if it has more than one host interface.

Reply:
I would also agree to use default=True.